src/test/java/fr/sncf/osrd/infra_state/RouteStateTest.java

```diff
@@ -550,6 +554,81 @@ public void testCircularInfraReserves() {
         }
     }
 
+    @Test
+    public void testTinyInfra2Trains() throws InvalidInfraException {
+        final var config = TestConfig.readResource("tiny_infra/config_railjson_2trains.json");
+        
+        var infra = config.rjsInfra;
+        var switchID = infra.switches.stream().findFirst().get().id;
+        var trainOrderedList = new ArrayList<String>();
+        trainOrderedList.add("First");
+        trainOrderedList.add("Second");
+
+        var initTable = new RJSSuccessionTable(switchID, trainOrderedList.toArray(new String[2]));
+        config.rjsSuccessions = new RJSSuccessions(Collections.singletonList(initTable));
+
+        var simState = config.prepare();
+        simState.run();
+        var sim = simState.sim;
+        var secondTrain = sim.trains.get("Second");
+        var finalPosition = secondTrain.getLastState().location.getPathPosition();
+        var initialPosition = Train.getInitialLocation(secondTrain.schedule, sim).getPathPosition();
+
+        // Check that the second train moves
+        assertNotEquals(initialPosition, finalPosition);
+    }
+
+    @Test
+    public void testChangeSuccessionTable() throws InvalidInfraException {
+        final var config = TestConfig.readResource("tiny_infra/config_railjson_2trains.json");
+
+        var switchID = config.rjsInfra.switches.stream().findFirst().get().id;
+        var trainOrderedList = new ArrayList<String>();
+        trainOrderedList.add("First");
+        trainOrderedList.add("Second");
+
+        var initTable = new RJSSuccessionTable(switchID, trainOrderedList.toArray(new String[2]));
+        config.rjsSuccessions = new RJSSuccessions(Collections.singletonList(initTable));
+
+        var simState = config.prepare();
+        var sim = simState.sim;
+
+        // Change the succession Table
+        var newTrainOrderedList = new ArrayList<String>();
+        newTrainOrderedList.add("Second");
+        newTrainOrderedList.add("First");
+        sim.infraState.towerState.changeTable(newTrainOrderedList, switchID);
+
+        simState.run();
+
+        var secondTrain = sim.trains.get("Second");
+        var firstTrain = sim.trains.get("First");
+        // Check that second train arrives first
+        assert secondTrain.getLastState().time < firstTrain.getLastState().time;
+    }
+
+    @Test
+    public void testChangeSuccessionTableDuringSim() throws InvalidInfraException {
+        final var config = TestConfig.readResource("tiny_infra/config_railjson_2trains.json");
+
+        var switchID = config.rjsInfra.switches.stream().findFirst().get().id;
+
+        var simState = config.prepare();
+        var sim = simState.sim;
+
+        // Change the succession Table
+        var newTrainOrderedList = new ArrayList<String>();
+        newTrainOrderedList.add("Second");
+        newTrainOrderedList.add("First");
+        makeFunctionEvent(sim, 0, () -> sim.infraState.towerState.changeTable(newTrainOrderedList, switchID));
```

Comment:
```suggestion
        makeFunctionEvent(sim, 1, () -> sim.infraState.towerState.changeTable(newTrainOrderedList, switchID));
```

Reply:
In this way the second assert
makeAssertEvent(sim, 6, () -> sim.infraState.towerState.isCurrentAllowed(switchID, "Second"));
will not pass because for this specific case the route request is made at time = 0. 
So the first train will reserve before the succession table is modified.